sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AppendBlobItem.java

```diff
@@ -31,7 +32,9 @@
      * @param encryptionKeySha256 The encryption key used to encrypt the page blob.
      * @param blobAppendOffset The offset at which the block was committed to the block blob.
      * @param blobCommittedBlockCount The number of committed blocks in the block blob.
+     * @deprecated Please use the overload.
      */
+    @Deprecated
```

Comment:
We should check on our deprecation policy here. This strikes me as different from what we're doing with sas because over there, we are trying to actively move people away from an inferior workflow, hence deprecation. Here, it's not really a big deal if customers continue to use this overload because it's pretty innocuous and behaves as expected. That said, maybe we want to be proactive about paring down the number of overloads when new functionality is introduced. 

@alzimmermsft  can you weigh in here on what the guidelines are?

Reply:
I generally think of API as something you need to pay rent on. Each public method incurs some small cost - maintenance, testing, documentation, samples, and cognitive load for end users. It's vanishingly small, but not zero cost. 

I don't want us to be in the business of aggressively deprecating anything we no longer like the look of, but I also don't want us keeping around duplicative or redundant API if it is truly unnecessary for any particular use case. Eagerly applying the deprecated annotations in this case should result in less teeth gnashing later, as users will have had potentially many more releases of deprecation warnings before any such breaking changes are applied (when the deprecated API is removed). 

The final point I'll make is that I wouldn't want our libraries to become vast wastelands of deprecation, because that signals to users our libraries are not as stable as they would like, or that we haven't put in enough thought up front to have a library that will stand the tests of time.